### PR TITLE
SALTO-7076: Add lint rules to prevent invalid imports

### PR DESCRIPTION
**Pull request by ori-moisis**
Add rules to prevent importing from files that are not distributed to NPM

also fix turbo config to re-run lint if the eslint configuration is changed

---

_Additional context for reviewer_

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -3,35 +3,26 @@
...
-      ],
-      "dependsOn": [
-        "^build-ts"
-      ]
+      "outputs": ["dist/**"],
+      "dependsOn": ["^build-ts"]
     },
     "lint": {
       "outputLogs": "new-only",
+      "inputs": ["$TURBO_DEFAULT$", "eslint.config.mjs", "../../eslint.config.mjs"],
```
Comment (ori-moisis) [inline turbo.json:11]: this should be the only real change, the rest are just prettier doing prettier things

File: packages/core/test/common/workspace.ts [hunk 2]
```diff
@@ -149,3 +151,40 @@ export const mockWorkspace = ({
     }) => (staticFilesSource ? staticFilesSource.getStaticFile({ filepath, encoding, isTemplate, hash }) : undefined),
   } as unknown as workspace.Workspace
 }
+
+export const mockAdaptersConfigSource = (): MockInterface<AdaptersConfigSource> => {
```
Comment (ori-moisis) [inline packages/core/test/common/workspace.ts:155]: not super happy about it, but this is copied from `@salto-io/workspace/test/common/workspace`

File: packages/jira-adapter/src/dependency_changers/object_type_order_to_schema.ts
Comment (yelly) [inline packages/jira-adapter/src/dependency_changers/object_type_order_to_schema.ts:1]: Why is this here?

File: packages/jira-adapter/src/filters/layouts/layout_service_operations.ts
Comment (yelly) [inline packages/jira-adapter/src/filters/layouts/layout_service_operations.ts:1]: Why is this here?

**Review by yelly — APPROVED** [review]
Just a couple files that seem unrelated.

File: packages/jira-adapter/src/filters/layouts/layout_service_operations.ts
Comment (ori-moisis) [reply packages/jira-adapter/src/filters/layouts/layout_service_operations.ts:1]: Another conflict with https://github.com/salto-io/salto/pull/6957
this time the solution there was wrong since jira filters should return the jira filter result, not the "generic" one

File: packages/jira-adapter/src/dependency_changers/object_type_order_to_schema.ts
Comment (yelly) [reply packages/jira-adapter/src/dependency_changers/object_type_order_to_schema.ts:1]: Nah just making sure these didn't get in by accident
